packages/core/src/hydration/api.ts

```diff
@@ -65,6 +72,19 @@ function enableHydrationRuntimeSupport() {
   }
 }
 
+/**
+ * Brings in the necessary i18n hydration code in a tree-shakable manner.
+ *
+ * Similar to `enableHydrationRuntimeSupport`, bringing in this code only
+ * when `withI18nHydration` is invoked.
+ */
+function enableI18nHydrationRuntimeSupport() {
+  if (!_isI18nHydrationSupportEnabled) {
+    _isI18nHydrationSupportEnabled = true;
+    enableLocateOrCreateNodeImpl();
```

Comment:
nit: can we add `i18n` into the function name?
```suggestion
    enableLocateOrCreateI18nNodeImpl();
```